File: Easy/2778_Sum_of_Squares_of_Special_Elements/Solution.py
from typing import List, Optional


class Solution:
    def sumOfSquares(self, nums: List[int]) -> int:
        n = len(nums)
        ans = 0
        
        for i in range(n):
            if n % (i + 1) == 0:
                ans += nums[i] * nums[i]
        
        return ans


if __name__ == "__main__":
    solution = Solution()

    # test cases 1
    nums1 = [1, 2, 3, 4]
    print(solution.sumOfSquares(nums1))

    # test cases 2
    nums2 = [2, 7, 1, 19, 18, 3]
    print(solution.sumOfSquares(nums2))
    

File: Easy/2778_Sum_of_Squares_of_Special_Elements/Solution.java



public class Solution {
    public int sumOfSquares(int[] nums) {
        int n = nums.length;
        int ans = 0;

        for (int i = 0; i < n; i++) {
            if (n % (i + 1) == 0) {
                ans += nums[i] * nums[i];
            }
        }
        
        return ans;
    }

    public static void main(String[] args) {
        Solution solution = new Solution();

        // test cases 1
        int[] nums1 = {1,2,3,4};
        System.out.println(solution.sumOfSquares(nums1));

        // test cases 2
        int[] nums2 = {2,7,1,19,18,3};
        System.out.println(solution.sumOfSquares(nums2));
        
    }
}